File: MatchMe-tvOSGameObjC Cropped/MatchMe/MatchMe/LevelTableViewCell.m
//
//  LevelTableViewCell.m
//  MatchMe
//
//  Created by Siyana Slavova on 2/22/16.
//  Copyright © 2016 mycompany.MatchMe. All rights reserved.
//

#import "LevelTableViewCell.h"

@interface LevelTableViewCell ()
@property (weak, nonatomic) IBOutlet UILabel *levelNumberLabel;
@property (weak, nonatomic) IBOutlet UILabel *levelTitleLabel;
@property (weak, nonatomic) IBOutlet UILabel *levelDescriptionLabel;
@property (weak, nonatomic) IBOutlet UIImageView *levelImage1;
@property (weak, nonatomic) IBOutlet UIImageView *levelImage2;
@property (weak, nonatomic) IBOutlet UIImageView *levelImage3;

@end

@implementation LevelTableViewCell

- (void)awakeFromNib {
    self.contentView.layer.borderWidth = 4;
    self.contentView.layer.borderColor = [UIColor lightGrayColor].CGColor;
}

- (void)configureWithLevelNumber:(NSInteger)levelNumber title:(NSString *)levelTitle description:(NSString *)levelDescription imageNames:(NSArray *)imageNames {
    self.levelNumberLabel.text = [NSString stringWithFormat:@"%ld",(long)levelNumber];
    self.levelTitleLabel.text = levelTitle;
    self.levelDescriptionLabel.text = levelDescription;
    [self.levelImage1 setImage:[UIImage imageNamed:imageNames[0]]];    
    [self.levelImage2 setImage:[UIImage imageNamed:imageNames[1]]];
    [self.levelImage3 setImage:[UIImage imageNamed:imageNames[2]]];
    
    self.levelImage1.hidden = YES;
    self.levelImage2.hidden = YES;
    self.levelImage3.hidden = YES;
}

#pragma mark - Focus

- (BOOL)canBecomeFocused {
    return YES;
}

- (void)becomeOnFocus:(BOOL)onFocus {
    self.contentView.layer.opacity = onFocus ? 1 : 0.7;
    self.levelImage1.hidden = !onFocus;
    self.levelImage2.hidden = !onFocus;
    self.levelImage3.hidden = !onFocus;
}

@end

File: MatchMe-tvOSGame Cropped/MatchMe/LevelTableViewCell.swift
//
//  LevelTableViewCell.swift
//  MatchMe
//
//  Created on 10/24/15.
//  Copyright © 2015. All rights reserved.
//

import UIKit

class LevelTableViewCell: UITableViewCell {
    @IBOutlet weak var levelNumberLabel: UILabel!
    @IBOutlet weak var levelTitleLabel: UILabel!
    @IBOutlet weak var levelDescriptionLabel: UILabel!

    @IBOutlet weak var levelImage1: UIImageView!
    @IBOutlet weak var levelImage2: UIImageView!
    @IBOutlet weak var levelImage3: UIImageView!

    override func awakeFromNib() {
        super.awakeFromNib()
        self.contentView.layer.borderWidth = 4
        self.contentView.layer.borderColor = UIColor.lightGrayColor().CGColor
    }
    
    override func canBecomeFocused() -> Bool {
        return true
    }
    
    func configureWith(number:String, title:String, description:String, imageNames:[String]) -> Void {
        self.levelNumberLabel.text = number
        self.levelTitleLabel.text = title
        self.levelDescriptionLabel.text = description
        self.levelImage1.image = UIImage(named: imageNames[0])
        self.levelImage2.image = UIImage(named: imageNames[1])
        self.levelImage3.image = UIImage(named: imageNames[2])
    }
    
    func becomeOnFocus(onFocus:Bool) -> Void {
        self.contentView.layer.opacity = onFocus ? 1 : 0.7
        self.levelImage1.hidden = !onFocus
        self.levelImage2.hidden = !onFocus
        self.levelImage3.hidden = !onFocus

    }
}